api/agent.go

```diff
@@ -7,14 +7,22 @@ import (
 
 // AgentCheck represents a check known to the agent
 type AgentCheck struct {
-	Node        string
-	CheckID     string
-	Name        string
-	Status      string
-	Notes       string
-	Output      string
-	ServiceID   string
-	ServiceName string
+	Node                           string
+	CheckID                        string
+	Name                           string
+	Status                         string
+	Notes                          string
+	Output                         string
+	ServiceID                      string
+	ServiceName                    string
+	HTTP                           string
+	Header                         map[string][]string
+	Method                         string
+	TLSSkipVerify                  bool
+	TCP                            string
+	Interval                       ReadableDuration
+	Timeout                        ReadableDuration
+	DeregisterCriticalServiceAfter ReadableDuration
```

Comment:
This should have the `Definition` sub-structure.

Reply:
Or actually is this needed at all?

Reply:
this is for registering checks on the local agent endpoint, so it shouldn't get the definition substructure